### PR TITLE
Health management

### PR DESCRIPTION
**Pull request by Snurrigt1**
- Health management
- Health bar
- Enemy damage rate

### CODE REVIEW

File: Player/player.tscn [hunk 1]
```diff
@@ -1,10 +1,19 @@
...
+[gd_scene load_steps=7 format=3 uid="uid://cwibpj61yuhlv"]
 
 [ext_resource type="Script" path="res://Player/player.gd" id="1_62xea"]
 [ext_resource type="Texture2D" uid="uid://cw7qc4s44biy6" path="res://Player/character_robot_walk0.png" id="2_uka3s"]
 
 [sub_resource type="RectangleShape2D" id="RectangleShape2D_milye"]
-size = Vector2(43, 69.75)
+size = Vector2(43, 48.875)
+
+[sub_resource type="RectangleShape2D" id="RectangleShape2D_1arvl"]
```
Comment (xix-xeaon) [inline Player/player.tscn:9]: I believe we'll need the collisionshape of the hurtbox to extend beyond the collisionshape of the character or it won't detect any enemies.

File: Player/player.gd [hunk 2]
```diff
@@ -1,9 +1,19 @@
...
+
+var health = 100.0
 
 func _physics_process(delta):
 	var direction = Input.get_vector("move_left", "move_right", "move_up", "move_down")
 	velocity = direction * 600
 	move_and_slide()
-
+	
+	const DAMAGE_RATE = 5.0
```
Comment (xix-xeaon) [inline Player/player.gd:12]: Constants like this are usually put near the top of a file for easy access.

File: Player/player.tscn [hunk 3]
```diff
@@ -15,5 +24,26 @@ scale = Vector2(0.96875, 0.992188)
...
 
 [node name="CollisionShape2D" type="CollisionShape2D" parent="."]
-position = Vector2(60, 78)
+position = Vector2(60, 88.4375)
 shape = SubResource("RectangleShape2D_milye")
+
+[node name="Hurtbox" type="Area2D" parent="."]
+unique_name_in_owner = true
+collision_layer = 0
+collision_mask = 2
```
Comment (xix-xeaon) [inline Player/player.tscn:33]: If we're going to use collision layers then they'll need to be named or it will be difficult to keep track of them.

**Review by xix-xeaon — CHANGES_REQUESTED** [review]
Please cleariy in the PR how you intend the damage/death system to work and the signal. Same thing about collision layers.

**Comment by Snurrigt1** [comment]
The use of the signal health depleted will be used to make the game over screen appear in the future (when the game over screen has been implemented).

File: project.godot [hunk 4]
```diff
@@ -41,6 +41,10 @@ move_down={
 ]
 }
 
+[layer_names]
+
+2d_physics/layer_2="Player"
```
Comment (xix-xeaon) [inline project.godot:46]: It looks like the player is actually on layer 1. Shouldn't layer 2 be for enemies? That's the layer being detected by the hurtbox.

**Review by xix-xeaon — APPROVED** [review]

**Pull request merged**